provider/provider.go

```diff
@@ -0,0 +1,41 @@
+/* provider is a Terraform provider for MAAS.
+ */
+package provider
+
+import (
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/roblox/terraform-provider-maas/gmaw"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func Provider() terraform.ResourceProvider {
+	return &schema.Provider{
+		Schema: map[string]*schema.Schema{
+			"api_key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The MAAS API key",
+			},
+			"api_url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The MAAS server URL (eg http://1.2.3.4:80/MAAS)",
```

Comment:
FWIW can you make the default port here 5240 as that would match what the actual default port is installed at. It will be a useful reminder.